pkg/ocmagenthandler/ocmagenthandler_deployment.go

```diff
@@ -242,6 +271,18 @@ func deploymentConfigChanged(current, expected *appsv1.Deployment, log logr.Logg
 		} else if curImage != expImage {
 			changed = true
 		}
+
+		// Compare readiness probe change
+		if !reflect.DeepEqual(curReadinessProbe, expReadinessProbe) {
```

Comment:
`DeepEqual` is not effective here, at least not right now, as it will always fail and lead to OAO patching the deployment on every reconcile.

The reason is that `successThreshold` and `timeoutSeconds` are not currently not being specified, so they will be set with defaults (1, respectively), and the presence of those defaults on-cluster will throw off the DeepEqual check.

You can overcome this by setting successThreshold and timeoutSeconds in the deployment in this code.

Reply:
Same goes for the liveness probe as well.